lib/collection/src/operations/conversions.rs

```diff
@@ -132,6 +138,165 @@ impl From<Record> for api::grpc::qdrant::RetrievedPoint {
     }
 }
 
+impl TryFrom<api::grpc::qdrant::RetrievedPoint> for Record {
+    type Error = Status;
+
+    fn try_from(retrieved_point: api::grpc::qdrant::RetrievedPoint) -> Result<Self, Self::Error> {
+        Ok(Self {
+            id: retrieved_point.id.unwrap().try_into()?,
+            payload: Some(proto_to_payloads(retrieved_point.payload)?),
+            vector: Some(retrieved_point.vector),
+        })
+    }
+}
+
+impl TryFrom<api::grpc::qdrant::GetCollectionInfoResponse> for CollectionInfo {
+    type Error = Status;
+
+    fn try_from(
+        collection_info_response: api::grpc::qdrant::GetCollectionInfoResponse,
+    ) -> Result<Self, Self::Error> {
```

Comment:
I understand this is still WIP, so you might want to split this function as it is rather big right now.